src/Datadog.Trace/HttpOverStreams/DatadogHttpClient.cs

```diff
@@ -57,27 +58,67 @@ private HttpResponse ReadResponse(Stream responseStream)
             const int statusCodeStart = 9;
             const int statusCodeEnd = 12;
             const int startOfReasonPhrase = 13;
+            const int bufferSize = 10;
 
             // TODO: Get this from StringBuilderCache after we determine safe maximum capacity
             var stringBuilder = new StringBuilder();
 
-            var chArray = new byte[1];
-            void GoNextChar()
+            var chArray = new byte[bufferSize];
+
+            async Task GoNextChar()
             {
-                streamPosition++;
-                chArray[0] = (byte)responseStream.ReadByte();
+                await responseStream.ReadAsync(chArray, offset: 0, count: 1).ConfigureAwait(false);
```

Comment:
If we are at the end of the stream, `ReadAsync()` will return `0` and we will probably set `currentChar` to whatever previous value was in `chArray[0]`.

Reply:
Yeah, that's the same as the previous behaviour I think - I guess we can probably throw if we expect to get bytes but don't get any?

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1218/commits/656a470756c5926b3d68df33e3bfa4313e44babc. Throws if we unexpectedly reach the end of the stream (won't happen with NetworkStream, as that will either hang or throw)